Dolia1903 flask lesson

Dolia1903 opened the pull request:
dz3_flask_lesson

Dolia1903 committed:
Changed README.md

Dolia1903 committed:
Changed README.md

Dolia1903 committed:
Added dz_3 file main.py

mpashko commented on main.py:
```diff
@@ -0,0 +1,57 @@
+
+
+if __name__ == '__main__':
+    history_file = "history.html"
```
это судя по всему лишнее

mpashko commented on main.py:
```diff
@@ -0,0 +1,57 @@
+    converted_amount = amount * rates["PHP"]
+    f = open("history.html", "a")
+    f.writelines(str(["PHP", rates["PHP"], amount, converted_amount]))
+    f.close()
```
html формат для такой задачи чересчур. Достаточно было просто записывать данные в txt файл.

И у тебя во всех 3-х методах дублируется логика: выполнить запрос, сделать расчёт и записать в файл. Если есть такая копипаста то это всегда индикатор того, что эта логика должна быть реализована в отдельных функциях/классах

mpashko commented on main.py:
```diff
@@ -0,0 +1,57 @@
+    converted_amount = amount * rates["GBP"]
+    f = open("history.html", "a")
+    f.writelines(str(["GBP", rates["GBP"], amount, converted_amount]))
+    f.close()
```
лучше использовать `with open()`, тогда не надо помнить про закрытие файла

mpashko commented:
Надо было ещё использовать шаблон для отображения истории

Dolia1903 committed:
Added updated file main.py via upload 

- убраны лишние куски кода
- организованы отдельные функции для записи в файл и получения ответа на get запрос
- заменил файл истории с html на txt
- использовал шаблон для отображения истории (добавил и его, ведь он, вероятно, нужен для проверки) - temp_1.html

mpashko commented on main.py:
```diff
@@ -0,0 +1,59 @@
+    return 'Hello, World!'
+
+
+def write_to_file(*args):
```
не стоит лишний раз использовать *args, это имеет смысл делать когда ты действительно не знаешь какие аргументы будут приходить.
Почему не стоит? Ты вносишь неясность в код, и работать с ним становиться тяжелее, потому что приходиться дополнительно разбираться что же там может быть.
Поэтому когда это возможно всегда старайся указывать конкретно какие аргументы ожидаются.

mpashko commented on main.py:
```diff
@@ -0,0 +1,59 @@
+def write_to_file(*args):
+    with open('history.txt', 'a') as f:
+        list_to_write = [str(x) for x in [*args]]
+        if len(list_to_write) == 4:
```
если бы использовал не args этой проверки не понадобилось бы

mpashko commented on main.py:
```diff
@@ -0,0 +1,59 @@
+    with open('history.txt', 'a') as f:
+        list_to_write = [str(x) for x in [*args]]
+        if len(list_to_write) == 4:
+            f.write(', '.join(list_to_write))
```
`f.write(', '.join(list_to_write) + '\n')`
можно сразу так
